packages/geoprocessing/src/toolbox/geoblaze/rasterStats.ts

```diff
@@ -83,31 +83,44 @@ export const rasterStats = async (
 
   try {
     // If categorical raster, use histogram to calculate valid cells
-    statsByBand = categorical
-      ? (
-          await geoblaze.histogram(raster, projectedFeat, {
-            scaleType: "nominal",
-          })
-        ).map((h: any) => {
+    if (categorical) {
+      const histogram = await geoblaze.histogram(raster, projectedFeat, {
+        scaleType: "nominal",
+      });
+
+      // If no overlap
+      if (!histogram.length) {
```

Comment:
This doesn't seem to be a sufficient check, the new test case you added returns `[{}]` and seemed to pass whether running the old code or this new code.